utility.go

```diff
@@ -142,11 +146,39 @@ func getMaxConcurrency(key string, default_value int) int {
 			log.Panic("Unknown concurrency number ", err)
 		}
 	} else {
-		if default_value > 0 {
-			con = default_value
+		if defaultValue > 0 {
+			con = defaultValue
 		} else {
 			con = 10
 		}
 	}
 	return max(con, 1)
 }
+
+// CheckType grabs the file extension from PATH.
+func CheckType(path string) string {
```

Comment:
It's not CheckType() it's GetExtensions()
chances are regex can be part of the global state (static)

Reply:
Fixed